Content.Client/Flash/FlashOverlay.cs

```diff
@@ -49,21 +50,27 @@ public void ReceiveFlash(double duration)
             _lastsFor = duration;
         }
 
-        protected override void Draw(in OverlayDrawArgs args)
+        protected override bool BeforeDraw(in OverlayDrawArgs args)
         {
             if (!_entityManager.TryGetComponent(_playerManager.LocalEntity, out EyeComponent? eyeComp))
-                return;
-
+                return false;
             if (args.Viewport.Eye != eyeComp.Eye)
-                return;
+                return false;
 
-            var percentComplete = (float) ((_gameTiming.CurTime.TotalSeconds - _startTime) / _lastsFor);
-            if (percentComplete >= 1.0f)
-                return;
+            _percentComplete = (float) ((_gameTiming.CurTime.TotalSeconds - _startTime) / _lastsFor);
+            if (_percentComplete >= 1.0f)
+            {
+                _screenshotTexture = null;
+                return false;
+            }
+            return true;
+        }
```

Comment:
Doing basic "Can I draw" checks should not be modifying the state internally hidden away, iyou should be doing this in the part that actually gets the texture and not just bandaiding it later on in the tick.

Reply:
I am not sure how you want this changed. The get-texture part is done in the ReceiveFlash function, which is called once when is mob is getting flashed. The _percentComplete calculation has to be done each frame. [DrunkOverlay.cs](https://github.com/space-wizards/space-station-14/blob/master/Content.Client/Drunk/DrunkOverlay.cs) does this identically with _visualScale. I could add a FrameUpdate function like [RainbowOverlay.cs](https://github.com/space-wizards/space-station-14/blob/master/Content.Client/Drugs/RainbowOverlay.cs) does, but I am not sure if this would not be worse for performance. Note that the FlashOverlay is always active (see [here](https://github.com/space-wizards/space-station-14/blob/master/Content.Client/Entry/EntryPoint.cs)) and should probably be moved into a status effect coupled to a component like DrunkOverlay and RainbowOverlay do.

Reply:
However, I could move the null check for _screenshotTexture from Draw to BeforeDraw as well, since Draw is only called if BeforeDraw returns true.

Reply:
I now reworked the whole thing to use the status effect syystem and added a FrameUpdate function to the overlay. Please re-review.